### PR TITLE
Pin the colorama version only for Python 3.4

### PR DESCRIPTION
**Pull request by nightlark**
This PR will pin the colorama version to `<=0.4.1` only for Python 3.4 and allow newer versions of Python to continue receiving colorama updates. Follows discussion in #6340 about newer versions of Python still being able to use newer colorama releases.

- [ ] Target the `master` branch for bug fixes, documentation updates and trivial changes.
- [ ] Target the `features` branch for new features, improvements, and removals/deprecations.
- [x] Include documentation when adding new features.
- [x] Include new tests or update existing tests when applicable.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "atomicwrites>=1.0",
     'funcsigs>=1.0;python_version<"3.0"',
     'pathlib2>=2.2.0;python_version<"3.6"',
-    'colorama<=0.4.1;sys_platform=="win32"',
+    'colorama<=0.4.1;sys_platform=="win32";python_version=="3.4"',
+    'colorama;sys_platform=="win32";python_version>"3.4"',
```
Comment (RonnyPfannschmidt) [inline setup.py]: this is missing python 2.7

Comment (nightlark) [reply setup.py]: Would `python_version<="2.7",>"3.4"` be okay, or would `<"3"` be a better for handling python 2.7?

Given Python 2 is EOL in under a month, a 2.8 version shouldn't appear...

Comment (nicoddemus) [reply setup.py]: `python_version<="2.7",>"3.4"` seems right to me. 👍

Comment (RonnyPfannschmidt) [reply setup.py]: @nightlark this branch keeps 2.7 support, mainline already dropped the old pythons

Comment (The-Compiler) [reply setup.py]: Why 2.7, given that colorama 0.4.3 still runs there?!

Comment (nicoddemus) [reply setup.py]:
```python
    'colorama<=0.4.1;sys_platform=="win32";python_version=="3.4"',
    'colorama;sys_platform=="win32";python_version!="3.4"',
```

?

Comment (nightlark) [reply setup.py]: `python_version!="3.4"` seems reasonable given that lower python 3.x versions were never supported in the 4.x releases.

Comment (nightlark) [reply setup.py]: Okay, made the change (and fixed the syntax to use multiple environment markers for the conditions).

Comment (nicoddemus) [reply setup.py]: > Do you want the new commit squashed into the old one for a cleaner history?

Yes, but took the liberty of doing it myself. 😁 

Thanks for the PR and patience! 👍

**Pull request merged**